rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -431,16 +509,63 @@ def _build_tf_train_graph(self) -> Tuple["tf.Tensor", "tf.Tensor"]:
             self.scale_loss,
         )
 
-    def _build_tf_pred_graph(
-        self, session_data: "train_utils.SessionData"
-    ) -> "tf.Tensor":
+    def combine_sparse_dense_features(
+        self,
+        batch: Dict[Text, tf.Tensor],
+        key_prefix: Text,
+        feature_dim: int,
+        batch_size: int,
+    ):
+        key_sparse = f"{key_prefix}sparse"
+        key_dense = f"{key_prefix}dense"
+
+        # apply mean/sum to convert sequence to sentence features
+
+        if key_dense in batch and key_sparse in batch:
+            _sparse = tf.math.reduce_sum(
+                train_utils.tf_dense_layer(
+                    batch[key_sparse],
+                    feature_dim,  # TODO define proper size
+                    "a",
+                    self.C2,
+                    feature_dim=feature_dim,
+                    batch_size=batch_size,
+                )
+            )
+            _dense = tf.math.reduce_mean(batch[key_dense], axis=1)
+
+            return tf.concat([_sparse, _dense])
+
+        if key_dense in batch:
+            return tf.math.reduce_mean(batch[key_dense], axis=1)
+
+        if key_sparse in batch:
+            return tf.math.reduce_sum(
+                train_utils.tf_dense_layer(
+                    batch[key_sparse],
+                    feature_dim,  # TODO define proper size
+                    "a",
+                    self.C2,
+                    feature_dim=feature_dim,
+                )
+            )
+
+    def _build_tf_pred_graph(self, session_data: "SessionData") -> "tf.Tensor":
+        num_text_features = self._get_num_of_features(session_data, "text_features_")
+        num_intent_features = self._get_num_of_features(
+            session_data, "intent_features_"
+        )
+
         self.a_in = tf.placeholder(
-            tf.float32, (None, session_data.X.shape[-1]), name="a"
+            tf.float32, (None, None, num_text_features), name="a"
```

Comment:
effectively we might have two `a_in` and `b_in`, we need to simulate `batch` as placeholders